Sources/PubNub/Events/New/EventEmitter.swift

```diff
@@ -0,0 +1,129 @@
+//
+//  EventEmitter.swift
+//
+//  Copyright (c) PubNub Inc.
+//  All rights reserved.
+//
+//  This source code is licensed under the license found in the
+//  LICENSE file in the root directory of this source tree.
+//
+
+import Foundation
+
+// MARK: - StatusEmitter
+
+/// A protocol for types that emit PubNub status events from the Subscribe loop.
+public protocol StatusEmitter: AnyObject {
+  /// A closure to be called when the connection status changes.
+  var onConnectionStateChange: ((ConnectionStatus) -> Void)? { get set }
+  /// A closure to be called when a subscription error occurs.
+  var onSubscribeError: ((PubNubError) -> Void)? { get set }
```

Comment:
Correct, that's because of the old implementation. You're right, it should be enough to read an underlying error from `ConnectionStatus` enum like `connectionError` and `disconnectedUnexpectedly` and remove `onSubscribeError` closure you pointed out. So what I would suggest is improving existing ConnectionStatus like this:

```
enum ConnectionStatus {
  case connectionError(PubNubError)
  case disconnectedUnexpectedly(PubNubError)
}
```

The reason I haven't done it is that there was `disconnectedUnexpectedly` case introduced a long time ago before EE. Now it's time to change it even if it might be a breaking change. I will run it by the team.